packages/core/modules-sdk/src/medusa-app.ts

```diff
@@ -420,53 +421,24 @@ async function MedusaApp_({
     customRemoteFetchData: remoteFetchData,
   })
 
-  /**
-   * Query wrapper to provide specific API's and pre processing around remoteQuery.query
-   * @param queryConfig
-   * @param options
-   */
-  async function query<const TEntry extends string>(
-    queryConfig: RemoteQueryObjectConfig<TEntry>,
+  const query: RemoteQueryFunction = async (
+    queryOptions:
+      | RemoteQueryObjectConfig<any>
+      | RemoteQueryObjectFromStringResult<any>
+      | RemoteJoinerQuery,
```

Comment:
Decided to type the query function and remove overloads from the implementation. This shouldn't impact the typings. 

But have to discuss if you think otherwise